stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarDrawable.kt

```diff
@@ -0,0 +1,133 @@
+package io.getstream.chat.android.ui.avatar
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.ColorFilter
+import android.graphics.Paint
+import android.graphics.PixelFormat
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
+import android.media.ThumbnailUtils
+
+internal class AvatarDrawable(bitmaps: List<Bitmap>) : Drawable() {
+    private val avatarBitmaps = bitmaps.take(AvatarView.MAX_AVATAR_SECTIONS)
+    private val paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        isAntiAlias = true
+        isFilterBitmap = true
+        isDither = true
+    }
+    private var avatarItems: List<AvatarItem> = listOf()
+
+    override fun draw(canvas: Canvas) {
+        avatarItems.forEach {
+            canvas.drawBitmap(it.bitmap, bounds, it.position, paint)
+        }
+    }
+
+    override fun setAlpha(alpha: Int) {
+        paint.alpha = alpha
+    }
+
+    override fun onBoundsChange(bounds: Rect) {
+        super.onBoundsChange(bounds)
+        reconfigureItems()
+    }
+
+    override fun getOpacity() = PixelFormat.TRANSLUCENT
+
+    override fun setColorFilter(colorFilter: ColorFilter?) {
+        paint.colorFilter = colorFilter
+    }
+
+    private fun reconfigureItems() {
+        avatarItems = when (avatarBitmaps.size) {
```

Comment:
Cant we use something like: `avatarBitmap.take(4)` and a generic method to configure them? :thinking: 
"Configure methods" seems really similar, I am not sure if we could extract some "common part" there.
Some ideas:
-. Bottom view is almost the same when 3 or 4 avatars
-. Top View is almost the same when 2 or 3 avatars
-. Top View Corner is always `0, 0`
-.View hieght is always `full-hieght` or `height/2`
-. View width is always `full-width` or `widht/2`

Reply:
Simplified the class a bit and removed duplication